src/OnlineUpdateDialogs.ycp

```diff
@@ -357,4 +359,81 @@ some of them might not work as expected.");
 	else return false;
     }
 
+    // Returns all installed products
+    list<map> installed_products()
+    {
+        list<map> products = Pkg::ResolvableProperties("", `product, "");
+
+        return (list<map>)filter(map p, products, {
+            return p["status"]:nil == `installed;
+        });
+    }
+
+    // Returns all products over the end of life support
+    list<map> eol_products()
+    {
+        integer time_now = time();
+        list<map> products = installed_products();
+
+        return (list<map>)filter(map p, products, {
+            return p["eol"]:nil != nil && tointeger(p["eol"]:"0") < time_now;
```

Comment:
I don't know exactly the Pkg::Bindings implementation, but what if `p["eol"]` is zero? That would report this product as EOLed, but it would most probably _NOT_ have that meaning when defined in metadata this way.